### ISSUE TITLE
Mapnik: render icons on polygon POIs

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2129]
**[Submitted to the original trac issue database at 9.35am, Tuesday, 4th August 2009]**

Currently no icons are rendered on POIs which are entered as polygons, though they might get a different color, therefore many mappers enter a polygon AND a node, which should not happen (according to http://wiki.openstreetmap.org/wiki/Good_practice).

This is a patch to render icons and texts on polygons as well.

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 2.54am, Monday, 10th August 2009]**

Slightly cleaned up, now in r16952

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 12.46am, Wednesday, 19th August 2009]**

It's now rendering. Thanks for the patch!

**Comment by tomhughes** [comment]
**Author: fiveisalive**
**[Added to the original trac issue at 3.46am, Saturday, 29th August 2009]**

When this went live, it correctly rendered names as well as icons on polygon ways, such as libraries or churches, e.g.

Robbins Library in Arlington, MA, USA:

http://www.openstreetmap.org/browse/way/31495194

But I just converted a library point to a building and it rendered the icon correctly but seemed to omit the name, examples:

Cary Memorial Library in Lexington, MA, USA:

http://www.openstreetmap.org/browse/way/39710653

also a church:

http://www.openstreetmap.org/browse/way/39712986

has there been an additional change since this patch?

**Comment by tomhughes** [comment]
**Author: jburgess777@googlemail.com**
**[Added to the original trac issue at 9.29am, Monday, 31st August 2009]**

There was a bug in this patch which meant that the text-poly layer was not rendering. The issue is that the text style has a filter which references the aeroway column but this is not returned by the text-poly layer. This was fixed in osm.xml r17346.

I have just updated the live rendering with this fix so the tiles should be fixed when they are rendered again.
